x-pack/plugins/enterprise_search/server/lib/indices/delete_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestPutPipelineRequest } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+/**
+ * Response for deleting sub-pipeline from @ml-inference pipeline.
+ * If sub-pipeline was deleted successfully, 'deleted' field contains its name.
+ * If parent pipeline was updated successfully, 'updated' field contains its name.
+ */
+export interface DeleteMlInferencePipelineResponse {
+  deleted?: string;
+  updated?: string;
+}
+
+export const deleteMlInferencePipeline = async (
+  indexName: string,
+  pipelineName: string,
+  client: ElasticsearchClient
+) => {
+  const response: DeleteMlInferencePipelineResponse = {};
+  const parentPipelineId = `${indexName}@ml-inference`;
```

Comment:
non-blocking: Should we write a utility function for this so the same template can be used in the multiple places we need to build this id?